### PR TITLE
bpo-44409: Fix error location in tokenizer errors that happen during initialization

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue44409

### CODE REVIEW

File: Parser/pegen.c [hunk 1]
```diff
@@ -283,6 +283,8 @@ static void
 raise_tokenizer_init_error(PyObject *filename)
 {
     if (!(PyErr_ExceptionMatches(PyExc_LookupError)
+          || PyErr_ExceptionMatches(PyExc_SyntaxError)
+          || PyErr_ExceptionMatches(PyExc_UnicodeDecodeError)
```
Comment (lysnikolaou) [inline Parser/pegen.c]: This isn't needed, since it's checked below.

Comment (pablogsal) [reply Parser/pegen.c]: Whoops, bad copy paste :P

Comment (pablogsal) [inline Parser/pegen.c:287]:
```suggestion
          || PyErr_ExceptionMatches(PyExc_SyntaxError)
```

File: Parser/pegen.c [hunk 2]
```diff
@@ -283,6 +283,8 @@ static void
...
           || PyErr_ExceptionMatches(PyExc_ValueError)
           || PyErr_ExceptionMatches(PyExc_UnicodeDecodeError))) {
```
Comment (pablogsal) [inline Parser/pegen.c]:
```suggestion
    if (!(PyErr_ExceptionMatches(PyExc_LookupError)
          || PyErr_ExceptionMatches(PyExc_SyntaxError)
          || PyErr_ExceptionMatches(PyExc_ValueError)
          || PyErr_ExceptionMatches(PyExc_UnicodeDecodeError))) {
```

**Review by lysnikolaou — APPROVED** [review]
Yes!

**Pull request merged**